File: modules/server/eventdata/godmode.lua
Godmode = {}

if not ServerConfig.Modules.Godmode.enabled then
    return
end

function Godmode.ProcessEventData(sender, data)
    local rawTargetData = data["hitGlobalId"] or data["hitGlobalIds"][1]
    local ped, target = GetPlayerPed(sender), NetworkGetEntityFromNetworkId(rawTargetData)
    
    if DoesEntityExist(target) and IsPedAPlayer(target) then
        if GetPlayerInvincible(rawTargetData) then
            TriggerEvent("icarus:my602oxd71pv", sender, "Godmode [C3]", false, {})
        end
    end
end

File: modules/server/eventdata/godmode.js
module.exports = (sender, data) => {
    const netId = data.hitGlobalId || data.hitGlobalIds[0];
    const target = NetworkGetEntityFromNetworkId(netId);

    if (DoesEntityExist(target) && IsPedAPlayer(target) && GetPlayerInvincible(netId)) {
        emit("icarus:my602oxd71pv", sender, "Godmode [C5]", false);
        CancelEvent();
    }
};